src/parser.rs

```diff
@@ -2881,6 +2882,82 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// Parse a GRANT statement.
+    pub fn parse_grant(&mut self) -> Result<Statement, ParserError> {
+        let privileges = if self.parse_keyword(Keyword::ALL) {
+            GrantPrivileges::All { with_privileges_keyword: self.parse_keyword(Keyword::PRIVILEGES) }
+        } else {
+            GrantPrivileges::Privileges(
+                self.parse_comma_separated(Parser::parse_grant_permission)?
+                    .iter().map(|kw| match kw {
+                        Keyword::DELETE     => keywords::DELETE.into(),
```

Comment:
Please let me know if there is an easier way to stringify these 😅 

Actually, I'm guessing that a new explicit `GrantPrivilege` enum would be preferable here, as opposed to these strings?